components/bin/build

```diff
@@ -0,0 +1,318 @@
+ *
+ * @param {string} name       The path of the file relative to the root .ts directory
+ * @param {string[]} objects  Array of names of the object exported by the file
+ * @return {string[]}         Array of lines for the contents of the library file
+ */
+function processLines(file, objects) {
+    if (objects.length === 0) return [];
+    const dir = path.dirname(file).replace(/^\.$/, '');
+    const dots = dir.replace(/[^\/]+/g, '..') || '.';
+    const relative = path.join(dots, '..', MATHJAX3, dir, path.basename(file)).replace(/\.ts$/, '.js');
+    const name = path.parse(file).name;
+    const lines = [
+        '"use strict";',
+        `Object.defineProperty(exports, '__esModule', {value: true});`
+    ];
+    let source = (dir.replace(/\//g, '.') + '.' + name).replace(/^\./, '')
+        .replace(/\.[^.]*/g, (x) => (x.substr(1).match(/[^a-zA-Z_]/) ? '[\'' + x.substr(1) + '\']' : x));
+    if (exists(path.resolve(MATHJAX3, file.replace(/\.ts$/, '')))) source += '_ts';
+    for (const id of objects) {
+        lines.push(`exports.${id} = MathJax._.${source}.${id};`);
+    }
+    return lines;
+}
+
+/**
+ * @param {string} file     Path to a file
+ * @return {boolean}        True if the file exists, false if not
+ */
+function exists(file) {
+    if (!fs.existsSync(file)) return false;
+    //
+    // For case-insensitive filing systems (like Mac OS),
+    //   check that the file names match exactly
+    //
+    const [dir, name] = [path.dirname(file), path.basename(file)];
+    return fs.readdirSync(dir).indexOf(name) >= 0;
+}
+
+/**
+ * Recursively make any needed directories
+ *
+ * @param {string} dir   The directory relative to the library directory
+ */
+function makeDir(dir) {
+    const fulldir = path.resolve(LIB, dir);
+    if (fs.existsSync(fulldir)) return;
+    makeDir(path.dirname(fulldir));
+    fs.mkdirSync(fulldir);
+}
+
+/**
+ * Make a file in the library directory from the given lines of javascript
+ *
+ * @param {string} file     The name of the file relative to the root .ts directory
+ * @param {string[]} lines  The contents of the file to create
+ */
+function makeFile(file, lines) {
+    if (!lines.length) return;
+    const [dir, name] = [path.dirname(file), path.basename(file)];
+    makeDir(dir);
+    fs.writeFileSync(path.resolve(LIB, dir, name.replace(/\.ts$/, '.js')), lines.join('\n') + '\n');
+}
+
+/**
+ * Make the library file that adds all the exported objects into the MathJax global object
+ */
+function processGlobal() {
+    console.log(COMPONENT + '.ts');
+    const lines = [
+        '"use strict";',
+        `Object.defineProperty(exports, '__esModule', {value: true});`,
+        `require('${GLOBAL}').combineWithMathJax({_: {`
+    ];
+    const packages = [];
+    PACKAGE = PACKAGE.sort(sortDir);
+    while (PACKAGE.length) {
+        const dir = path.dirname(PACKAGE[0]).split(path.sep)[0];
+        packages.push(processPackage(lines, '    ', dir));
```

Comment:
I assume that is meant to produce indentation? Maybe make this a configurable `const` in the light of issue ?